Source/Serilog.Exceptions/Reflection/ReflectionInfoExtractor.cs

```diff
@@ -0,0 +1,122 @@
+namespace Serilog.Exceptions.Reflection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// Utility class that analyzes type using reflection and provides
+    /// information about properties to be used in destructuring process.
+    /// </summary>
+    internal class ReflectionInfoExtractor
```

Comment:
Just a thought. It looks like this could be made static. Maybe `GetOrCreateReflectionInfo` could be an extension method on `Type`?

Reply:
Because the cache is involved the class itself or `GetOrCreateReflectionInfo` cannot be static.
Some core internal "side-effect free" logic could be extracted, although such extraction feels unnatural to me right now. 
Of course, we could always refactor later :-).